tests/test_concurrency_async.py

```diff
@@ -45,7 +44,8 @@ async def runner():
 
 
 @pytest.mark.slow
-async def test_concurrent_execution(aconn_cls, dsn):
+@pytest.mark.parametrize("anyio_backend", ["asyncio"])
+async def test_concurrent_execution(aconn_cls, dsn, anyio_backend):
```

Comment:
I don't get exactly what is the reason for specifying this parametrize here. I understand that `anyio_backend` is a parametrized fixture that yields exactly one backend right now, right?

I understand that your plan is to eventually change the fixture and make it yield `['asyncio', 'trio']`, but that a test like this cannot run as it is on trio because it uses explicitly `asyncio.gather()`. That might be fine, but I think that pre-emptively skipping this test on trio might hide failures. I would rather not specify a `@parametrize` for this test, look at everything failing on `trio`, and only then write a matching test `test_concurrent_execution_trio`... Although that wouldn't scale in the direction of having different backends.

What do you think will be supported in the future?

Also, alternative implementation, to make tests more readable (although it would add the same shortcoming of hiding tests when a new backend is introduced): how about dropping the module-level `import asyncio`, and having a fixture called `asyncio`, which would yield the asyncio module if we are testing asyncio and would skip the tests if we are testing trio? Same for other backends.

```suggestion
async def test_concurrent_execution(aconn_cls, dsn, asyncio):
```

Reply:
Indeed, this `mark` is somehow a left-over which could be dropped. Namely, I initially did not port those tests (the ones using `asyncio` explicitly) and thus needed this `mark`; but ultimately, I decided to port them in https://github.com/psycopg/psycopg/pull/146/commits/f43c709e3e90ebe1701e0c564f19512fd648f192#diff-b3b8cfbbb6a3720571b3ceceb3dc998d151667f62e5738a225bd0234cbde51ff (done through `if anyio_backend_name == "asyncio":`, though we might arguably find a nicer way).